zipkin-storage/cassandra/src/main/java/zipkin2/storage/cassandra/internal/call/ResultSetFutureCall.java

```diff
@@ -76,16 +80,21 @@ protected final boolean doIsCanceled() {
   }
 
   static ResultSet getUninterruptibly(ListenableFuture<ResultSet> future) {
-    if (future instanceof ResultSetFuture) {
-      return ((ResultSetFuture) future).getUninterruptibly();
-    }
-    try { // emulate ResultSetFuture.getUninterruptibly
-      return Uninterruptibles.getUninterruptibly(future);
-    } catch (ExecutionException e) {
-      Throwable cause = e.getCause();
-      if (cause instanceof Error) throw ((Error) cause);
-      if (cause instanceof DriverException) throw ((DriverException) cause).copy();
-      throw new DriverInternalError("Unexpected exception thrown", cause);
+    try {
+      if (future instanceof ResultSetFuture) {
+        return ((ResultSetFuture) future).getUninterruptibly();
+      }
+      try { // emulate ResultSetFuture.getUninterruptibly
+        return Uninterruptibles.getUninterruptibly(future);
+      } catch (ExecutionException e) {
+        Throwable cause = e.getCause();
+        if (cause instanceof Error) throw ((Error) cause);
+        if (cause instanceof DriverException) throw ((DriverException) cause).copy();
+        throw new DriverInternalError("Unexpected exception thrown", cause);
+      }
+    } catch (QueryConsistencyException | BusyConnectionException | BusyPoolException e) {
+      // Our throttling function relies on rejected execution when encountering load-specific errors
+      throw new RejectedExecutionException(e.getMessage(), e);
```

Comment:
Mentioned this in a related review, but rather than stutter the exception message can we put something more descriptive in it?  E.g. "Cassandra overloaded; please enable storage throttling so we can attempt to slow request rate."

Reply:
on the line of stuttering :) this ends up in heath checks and 500 errors on read. having a relevant top-level message is important. Errors are not isolated to writes even if the code you are thinking about is.

Reply:
also storage throttling isn't the only way. for example, cassandra has its own backoff queue local to handle things like this. it is just allowing those who alternatively want to use this. I really don't think we should be advertizing storage throttling in expection messages as an answer as depending on the product different tools can be used, including better tuning.

Reply:
if we are going to monkey with messages, make them misleading, obfuscated etc, I'd rather refactor out the RejectedExecutionException signal. so that we don't have to

Reply:
just right now there's tons of work and little hands to help with it, it is much easier to use the source data as we do with elasticsearch already.

Ex below we can see the actual result in health checks or 500 errors resulting from browser etc.
```
        if (status != null && status.intValue() == 429) {
          toThrow = new RejectedExecutionException(tree.toString());
```

To change that here would just make troubleshooting harder